web/src/pages/feed.tsx

```diff
@@ -67,6 +61,13 @@ export default function Feed({ searchValue }: Props) {
     });
   }, [hasNextPage, isFetchingNextPage]);
 
+  // Cleanup and cancel subscriptions
+  useEffect(() => {
+    return () => {
+      debouncedSetSearchValue.cancel();
+    };
+  }, []);
```

Comment:
I'm nervous about the empty dependency array here because I think you really want `[debouncedSetSearchValue]`